include/proc_ui/procui.h

```diff
@@ -28,34 +57,96 @@ typedef uint32_t (*ProcUICallback)(void *);
 
 typedef enum ProcUICallbackType
 {
+   //! Application acquires the foreground
    PROCUI_CALLBACK_ACQUIRE,
+   //! Application must release the foreground
    PROCUI_CALLBACK_RELEASE,
+   //! Application must exit
    PROCUI_CALLBACK_EXIT,
+   //! Application may start using networking
    PROCUI_CALLBACK_NET_IO_START,
+   //! Application must stop using networking
    PROCUI_CALLBACK_NET_IO_STOP,
+   //! The user attempted to press the HOME button but was denied
    PROCUI_CALLBACK_HOME_BUTTON_DENIED,
 } ProcUICallbackType;
 
 typedef enum ProcUIStatus
 {
+   //! The application is in the foreground. All resources may be used.
    PROCUI_STATUS_IN_FOREGROUND,
+   //! The application is in the background, only limited resources are usable.
    PROCUI_STATUS_IN_BACKGROUND,
+   //! The application must relelase the foregound - see ProcUIDrawDoneRelease()
    PROCUI_STATUS_RELEASE_FOREGROUND,
+   //! The application must release all resources (including ProcUI) and quit
    PROCUI_STATUS_EXITING,
 } ProcUIStatus;
 
 uint32_t
 ProcUICalcMemorySize(uint32_t unk);
 
+/**
+ * Unregister all ProcUI callbacks.
+ *
+ * \sa
+ * - ProcUIRegisterCallback()
+ *
+ * \if false
+ * Would a currently executing async operation keep or lose the callbacks?
+ * does this block?
+ * \endif
+ */
 void
 ProcUIClearCallbacks();
 
+/**
+ * Signifies to ProcUI that the current application has released all foreground
+ * resources, drawn its last frame, and is ready to be moved into the
+ * background. Should only be called when the application is in the 
+ * #PROCUI_STATUS_RELEASE_FOREGROUND state.
+ *
+ * \note
+ * Upon calling this function, all application threads on cores 0 or 1 are
```

Comment:
Nothing actually happens until ProcUIProcessMessages is called again after this function has been called.

The code for ProcUIDrawDoneRelease is just:
```
ProcUIDrawDoneRelease:
lis       r12, dword_10000444@ha
li        r0, 1
stw       r0, dword_10000444@l(r12)
blr
```

dword_10000444 is loaded from in ProcUIProcessMessages where you will see it triggers the release process COSWarn("ProcUI: Releasing foreground\n").